### PR TITLE
WIP: allow assignment syntax in @strdict

### PR DESCRIPTION
**Pull request by JonasIsensee**
This PR is a working implementation for #239 

This is still WIP as it needs discussion e.g. what other macros could also benefit from this.

### CODE REVIEW

**Comment by Datseris** [comment]
So this allows parsing the following combinations right:
```julia
@dict a b c=some_existing_variable d e=some_other_var
```

**Comment by sebastianpech** [comment]
As this adds macrotools as a dependency, I guess those can be removed:
https://github.com/JuliaDynamics/DrWatson.jl/blob/2b96179159c519f1367433e65dc01900de0a4c31/src/dict_list.jl#L133-L136

**Comment by sebastianpech** [comment]
btw. MacroTools is not >= 1. I don't really care, but I remember a discussion about that ;)

**Comment by Datseris** [comment]
> btw. MacroTools is not >= 1

Nooooooooooooo how can this be!!!!!!!!!!!!!!!!!!!!!!!!!!! There should be a thousand packages that depend on it!!!!!!!!!!!!!!!!!!!

**Comment by sebastianpech** [comment]
> Nooooooooooooo how can this be!!!!!!!!!!!!!!!!!!!!!!!!!!! There should be a thousand packages that depend on it!!!!!!!!!!!!!!!!!!!

Knew it ;) ... Should've let @JonasIsensee sneak the macrotools dependency in

**Comment by tamasgal** [comment]
Oh yeah, the good old not-version-one-yet dilemma and more and more posts on Discord because packages won't upgrade due to pinned versions 🙈

**Comment by JonasIsensee** [comment]
Honestly, semver is a convention. There are never any guarantees.

`MacroTools` is a stable package. It's had a single release in the last 12 months and
from the PR history it does not appear to have had any breaking changes in the last 
2 years.

I vote to just depend on `MacroTools`. If they ever break something, we can still just fix it.

**Comment by Datseris** [comment]
okayyyyyyyyyyyyy okayyyy with a lot of saltiness we can use MacroTools. Jonas, is this correct: https://github.com/JuliaDynamics/DrWatson.jl/pull/240#issuecomment-807382361 ?

**Comment by JonasIsensee** [comment]
> okayyyyyyyyyyyyy okayyyy with a lot of saltiness we can use MacroTools. Jonas, is this correct: [#240 (comment)](https://github.com/JuliaDynamics/DrWatson.jl/pull/240#issuecomment-807382361) ?

Yes, that's correct. :)

File: src/naming.jl [hunk 1]
```diff
@@ -283,10 +284,18 @@ Same as [`@dict`](@ref) but the key type is `String`.
 """
 macro strdict(vars...)
```
Comment (Datseris) [inline src/naming.jl:285]: So my understanding now is that this works only for `@strdict`? Then it should also be done in `@dict, @ntuple, @savename`.

Comment (JonasIsensee) [reply src/naming.jl:285]: :+1: 
(yes, as I edited above) 
Will do, when I get the chance

**Review by Datseris — APPROVED** [review]
cool, also bump minor version in Project.toml please

**Pull request merged**